### PR TITLE
Prevents evidence bags holding more bags.

### LINKED ISSUE #49239: Evidence bags are F U C K
## Reproduction:
Try and put armory in evidence bags. Cry.
[Hell](https://media.discordapp.net/attachments/458452245256601615/675476297161637948/unknown.png?width=1089&height=681)
Try and put armory in one box. Cry.
[Hell 2](https://cdn.discordapp.com/attachments/458452245256601615/675477754199801938/unknown.png)
[See above](https://media.discordapp.net/attachments/458452245256601615/675477836320079935/unknown.png?width=1089&height=681)

This is with the roundstart evidence boxes on box. With 2 more you could fit everything in one box.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
Begone, fix-feature bot.

## About The Pull Request

This disallows evidence bags from holding any subtypes of obj/item/storage, so that evidence bags can't hold other storage items, that hold evidence bags holding storage items that hold evidence bags holding storage items etc. 

Ideally, they should probably be a storage subtype themselves, but this will keep them from holding effectively infinite storage.

## Why It's Good For The Game

Fixes #49239 . 

## Changelog
:cl:
fix: Evidence Bags can no longer hold items that store other items.
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
Semi blessed, sliding medium sized items into the bags while said bags are in boxes seems a little sus, not sure what can be done about that tho.

**Comment by ArcaneMusic** [comment]
I'm working on a better fix at the second so that it checks against loc's parent storage sizes right now.

**Comment by LemonInTheDark** [comment]
You've ripped my heart and det play apart. Blessed.

**Comment by cacogen** [comment]
I think I remember back in the day someone suggesting that evidence bags take on the w_class of their contents e.g. you can't put a box inside a box

**Comment by ninjanomnom** [comment]
Assigning me the pr what the fuck

File: code/__DEFINES/components.dm [hunk 1]
```diff
@@ -411,6 +411,7 @@
 #define COMSIG_TRY_STORAGE_QUICK_EMPTY "storage_quick_empty"			//(loc) - returns bool - if loc is null it will dump at parent location.
 #define COMSIG_TRY_STORAGE_RETURN_INVENTORY "storage_return_inventory"	//(list/list_to_inject_results_into, recursively_search_inside_storages = TRUE)
 #define COMSIG_TRY_STORAGE_CAN_INSERT "storage_can_equip"				//(obj/item/insertion_candidate, mob/user, silent) - returns bool
+#define COMSIG_RETURN_MAX_COMBINED_W "storage_max_combined_weight"		//() - returns number.
```
Comment (ninjanomnom) [inline code/__DEFINES/components.dm]: No number returns in signals allowed, you may only return flags

File: code/modules/detectivework/evidence.dm [hunk 2]
```diff
@@ -28,6 +28,15 @@
 	if(!istype(I) || I.anchored == 1)
 		return
 
+	if(istype(loc, /obj/item/storage))
```
Comment (ninjanomnom) [inline code/modules/detectivework/evidence.dm]: You shouldn't be istyping, that's what the component is for

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Comment by optimumtact** [comment]
These things honestly need a bullet.

PS: conflict

File: code/__DEFINES/components.dm (hunk 1 above)
Comment (ShizCalev) [reply code/__DEFINES/components.dm]: https://github.com/tgstation/tgstation/blob/9e34b3d6a15df071938932f46e5276eb9529ccab/code/__DEFINES/dcs/signals.dm#L349 ?

File: code/__DEFINES/dcs/signals.dm [hunk 3]
```diff
@@ -1,391 +1,392 @@
...
-	#define COMPONENT_ACTION_BLOCK_TRIGGER 1
-
-//Xenobio hotkeys
-#define COMSIG_XENO_SLIME_CLICK_CTRL "xeno_slime_click_ctrl"				//from slime CtrlClickOn(): (/mob)
-#define COMSIG_XENO_SLIME_CLICK_ALT "xeno_slime_click_alt"					//from slime AltClickOn(): (/mob)
-#define COMSIG_XENO_SLIME_CLICK_SHIFT "xeno_slime_click_shift"				//from slime ShiftClickOn(): (/mob)
-#define COMSIG_XENO_TURF_CLICK_SHIFT "xeno_turf_click_shift"				//from turf ShiftClickOn(): (/mob)
-#define COMSIG_XENO_TURF_CLICK_CTRL "xeno_turf_click_alt"					//from turf AltClickOn(): (/mob)
-#define COMSIG_XENO_MONKEY_CLICK_CTRL "xeno_monkey_click_ctrl"				//from monkey CtrlClickOn(): (/mob)
+// All signals. Format:
```
Comment (ShizCalev) [inline code/__DEFINES/dcs/signals.dm]: why is the entirety of this file changed? CRLF?

Comment (ArcaneMusic) [reply code/__DEFINES/dcs/signals.dm]: Gonna be honest here chief, I have no clue. I think it had to do with VSC getting upset with ninja's component change the other day, but as far as I can tell it didn't bjork anything in terms of functionality.

Comment (optimumtact) [reply code/__DEFINES/dcs/signals.dm]: undo it

File: code/__DEFINES/components.dm (hunk 1 above)
Comment (ninjanomnom) [reply code/__DEFINES/components.dm]: Just because other things do it doesn't mean it acceptable

**Pull request merged**